### PR TITLE
Log more data for PrevalidateShardPathIT#testCheckShards

### PR DESCRIPTION
**Pull request by arteam**
`PrevalidateShardPathIT#testCheckShards` still keeps failing on CI, so we need more logging.

* Add debug logging for `MasterService` events to verify that a cluster service update is triggered.
* Log explanation for stuck shards regardless whether they are located on node_2 in the cluster state.

See #104807

### CODE REVIEW

**Review by pxsalehi — APPROVED** [review]

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
@elasticmachine update branch

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateShardPathIT.java [hunk 1]
```diff
@@ -39,6 +40,10 @@
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
 public class PrevalidateShardPathIT extends ESIntegTestCase {
 
+    @TestLogging(
+        value = "org.elasticsearch.cluster.service.MasterService:DEBUG",
+        reason = "https://github.com/elastic/elasticsearch-serverless/pull/2068"
```
Comment (idegtiarenko) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateShardPathIT.java:45]: Should it link https://github.com/elastic/elasticsearch/issues/104807 instead?

Comment (arteam) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateShardPathIT.java:45]: Thank you for catching this!

**Comment by arteam** [comment]
@elasticmachine update branch

**Pull request merged**